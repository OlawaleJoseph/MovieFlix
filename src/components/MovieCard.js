import { Button } from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';

const MovieCard = ({ imgUrl, handleTrailerClick, name }) => (
  <div className="movieCard">
    <div className="inner">
      <div className="front">
        <img src={imgUrl} alt="Movie Poster" />
      </div>
      <div className="back">
        <Button
          variant="outlined"
          size="large"
          color="secondary"
          onClick={() => handleTrailerClick(name)}
        >
          Trailer
        </Button>
        <Button
          variant="outlined"
          size="large"
          color="secondary"
        >
          Details
        </Button>
      </div>
    </div>

  </div>
);

MovieCard.defaultProps = {
  imgUrl: '',
};

MovieCard.propTypes = {
  imgUrl: PropTypes.string,
  name: PropTypes.string.isRequired,
  handleTrailerClick: PropTypes.func.isRequired,
};

export default MovieCard;
