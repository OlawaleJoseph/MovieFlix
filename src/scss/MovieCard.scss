.movieCard {
  width: 250px;
  height: 320px;
  perspective: 1000px;
  cursor: pointer;

  .inner {
    position: relative;
    width: 100%;
    height: 100%;
    transform-style: preserve-3d;
    transition: transform 1s;
  }

  &:hover .inner {
    transform: rotateY(180deg);
    text-align: center;
  }

  .front,
  .back {
    position: absolute;
    width: 100%;
    height: 100%;
    backface-visibility: hidden;
  }

  .front {
    background-color: green;

    img {
      object-fit: cover;
    }
  }

  .back {
    background-color: #131313;
    transform: rotateY(180deg);
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
  }
}
