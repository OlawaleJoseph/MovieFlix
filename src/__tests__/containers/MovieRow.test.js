// import { Typography, Button } from '@material-ui/core';
import Row from '../../containers/MovieRow';
import MovieCard from '../../components/MovieCard';
import shallowWrapper from '../../testSetup';

describe('Movie Row', () => {
  const rowInfo = {
    title: 'Test',
    movies: [{
      id: 'id',
      imageUrl: 'imageurl',
    }],
  };

  const row = shallowWrapper(Row, rowInfo);

  test('should render Movie Row', () => {
    expect(row).toBeTruthy();
    expect(row).toMatchSnapshot();
  });

  test('should have a title', () => {
    const movieDescription = row.find('.section__heading');

    expect(movieDescription).toHaveLength(1);
    expect(movieDescription.text()).toEqual(rowInfo.title);
  });

  test('should render movies', () => {
    const movies = row.find(MovieCard);

    expect(movies).toHaveLength(rowInfo.movies.length);
  });
});
