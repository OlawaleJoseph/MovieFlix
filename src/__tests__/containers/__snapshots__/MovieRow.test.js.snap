// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Section should render Banner 1`] = `ShallowWrapper {}`;

exports[`Hero Section should render Movie Row 1`] = `ShallowWrapper {}`;

exports[`Movie Row should render Movie Row 1`] = `ShallowWrapper {}`;

exports[`row Section should render Movie Row 1`] = `ShallowWrapper {}`;
